### ISSUE TITLE
Aplications runner should allow all apps inside $PATH

### ISSUE THREAD

**Issue by mattboll** [fyne-io/fynedesk#111]
At home I've got fyne_demo in ~/go/bin ; intellij in ~/bin/idea everything is in my $PATH but I can't access it using the runner, we shouldn't need a terminal to launch them

**Comment by andydotxyz** [comment]
The runner looks for "installed applications" rather than binaries.
If you "fyne install ./cmd/fyne_demo" then it should appear in the app list.
/usr/share/applications/ is the important path.

Do you think we should scan $PATH as well and open found apps in a terminal window?

**Comment by mattboll** [comment]
I didn't know about fyne install, does it work with not go binaries ?

Nobody should require sudo to add an app to the runner, xfce and gnome scan $PATH and I find it useful.

We should not open them in a terminal window… unless it should be ran in a terminal ? I don't know how to guess, let me explain some scenario : 
- download intellij (get a .zip file) unzip it, make an alias to the bin in the PATH, the runner should open intellij window without a terminal even though the binary is a shell script
- run a command such as "top", it should open a terminal

**Comment by andydotxyz** [comment]
If all you have is a binary I think it has to open in a terminal as you have no way of knowing. When you install IntelliJ is there a .desktop file alongside the script or some other metadata?

**Comment by mattboll** [comment]
There is no .desktop file in intellij, nor with android studio and I don't find anything that looks like it but they do launch in gnome runner.

But you're right there must be some similar information because gnome can't find mongo3t 🤔

**Comment by andydotxyz** [comment]
Maybe it’s icons? Does it display an app icon? If so try to see if there is a link there

**Comment by mattboll** [comment]
The only info I found about icon is in the parent directory (the launcher is idea-IU-191.6707.61/bin/idea.sh)
```

$ cat idea-IU-191.6707.61/product-info.json
{
  "name": "IntelliJ IDEA",
  "version": "2020.1.1",
  "buildNumber": "201.7223.91",
  "productCode": "IU",
  "dataDirectoryName": "IntelliJIdea2020.1",
  "svgIconPath": "bin/idea.svg",
  "launch": [
    {
      "os": "Linux",
      "launcherPath": "bin/idea.sh",
      "javaExecutablePath": "jbr/bin/java",
      "vmOptionsFilePath": "bin/idea64.vmoptions",
      "startupWmClass": "jetbrains-idea"
    }
  ]

```

**Comment by andydotxyz** [comment]
And it's not present in `/.local/share/applications` either?

**Comment by andydotxyz** [comment]
On develop I have added ~/.local/share/applications which made a difference for me, GoLand had popped itself in there on an Arch install.

**Comment by mattboll** [comment]
Yes it should be there, awesome ! I'll test asap

**Comment by andydotxyz** [comment]
Seems like this could be closed now that we've tracked it down to local user configuration?

**Comment by mattboll** [comment]
discord, intellij, goland, android studio (last 3 are basically the same software) still don't show for me (linux debian) but it works for microsoft teams
May be it's a bug only for me… I'll try to look furthermore some day

**Comment by andydotxyz** [comment]
GoLand is working fine here, it installed to the local space. Is it still an env issue, or are you out of date on develop (it’s only about 5 commits back)?

**Comment by mattboll** [comment]
I've updated develop before testing. May be an env issue, I thought microsoft teams was on the local path but actually no, «bot framework emulator» is on my local path too and not appearing. I tried to unset XDG_DATA_DIRS or add my local path to it with no luck

**Comment by andydotxyz** [comment]
You might need to log out after the env change for it to get picked up?

**Comment by mattboll** [comment]
I don't know if it would be not good citizens to add some path, I didn't find how xfce or gnome work but may be we can close this issue as long as we know the problem, the main bug is in snapd and it's not a priority for fynedesk right now
